src/cmap/manager.rs

```diff
@@ -59,6 +63,46 @@ impl PoolManager {
         Ok(())
     }
 
+    pub(super) fn store_pinned(
+        &self,
+        connection: Connection,
+    ) -> std::result::Result<(), Connection> {
+        if let Err(request) = self
+            .sender
+            .send(PoolManagementRequest::StorePinned(connection))
+        {
+            let conn = request.0.unwrap_check_in();
+            return Err(conn);
+        }
+        Ok(())
+    }
+
+    pub(super) async fn take_pinned(&self, id: u32) -> Result<Connection> {
```

Comment:
(For my own understanding) what is the difference between taking a pinned connection and unpinning?

Reply:
Pinning a connection removes it from the normal connection pool and makes it available to retrieve by id (wrapped by `PinnedConnectionHandle`).  Taking a pinned connection is that retrieval; it enforces ownership mechanics at runtime by only allowing a connection to be taken by id if that connection had previously been pinned and dropped.  Unpinning returns the connection to the normal connection pool, either immediately (if nothing's using it) or when it gets dropped.

Hopefully the docstrings added will help make this clearer :)